### ISSUE TITLE
Conflict with the Walled Templates module 

### ISSUE THREAD

**Issue by SilvesTheRog** [chrisk123999/chris-premades#223]
I've encountered a problem with the Hunger of Hadar spell when I was testing it out. It works fine, but I couldn't move the tokens that were affected by the spell with my mouse. When I tried to drag them, the drag ruler module showed as if I was dragging something from the center of the spell area, but the tokens remained still. 

Turned out, it happens because of the Walled Templates module (https://github.com/caewok/fvtt-walled-templates). Disabling it fixed the problem.

I recorded a video to demonstrate the issue:
https://www.youtube.com/watch?v=NELuzWTHDiY

Would be nice to have compatibility between these two.

**Comment by SilvesTheRog** [comment]
Yes, it happens only when Walled Templates module is enabled. I even tried testing it by disabling everything except for CPR and its dependencies. And then enabling Walled Templates alone caused the issue again:

https://www.youtube.com/watch?v=ISRgDT13JbM

**Comment by txm3278** [comment]
oh I see, that looks like it might be something new with walled templates. I will take a look

**Comment by SilvesTheRog** [comment]
Thanks!
I also noticed that Spirit Guardians don't reduce enemy movement speed. Is it intended or is it another bug by Walled Templates?

![image](https://github.com/chrisk123999/chris-premades/assets/127493227/4a88c185-55bf-4496-ae9d-ec3eadb59f68)

**Comment by txm3278** [comment]
Ever since enhanced terrain layer stopped being updated there hasn't been a good way to do this. Could potentially be a feature request using Caewok's Terrain Mapper

**Comment by SilvesTheRog** [comment]
I see. Well, I added an aura that halves the movement speed of affected enemies to the spirit guardians effect, and it seems to work just fine, so I'll just stick to this workaround (:

**Comment by LukasPrism** [comment]
> I see. Well, I added an aura that halves the movement speed of affected enemies to the spirit guardians effect, and it seems to work just fine, so I'll just stick to this workaround (:

Note that's a decent workaround but doesn't really solve the issue. For instance a PC might move 20' before entering the template. At which point their 30' of movement *speed* is halved by your AE, equalling 15'. Drag ruler should(?) show they've used all their movement because they already spent 20 but now only "have" 15... but actually they would still have 5' left. Maybe DR is smart enough to figure this out, but last I checked it was a problem.

**Comment by caewok** [comment]
Try Walled Templates v0.7.6. BTW, I cannot guarantee compatibility with Drag Ruler. If you run into compatibility issues, try my Elevation Ruler module.

**Comment by SilvesTheRog** [comment]
Oh yes, it's working now! Thank you!

![image](https://github.com/chrisk123999/chris-premades/assets/127493227/c27ad55f-f18e-4840-8136-baa99872657b)

**Comment by NdranC** [comment]
I'm not sure if this is exactly related to my issue but I came here to post an issue I have between Walled Templates, Chris's Premades, Gambit's Premades and sightly Drag Ruler.

Here is a video showcasing my problem https://youtu.be/1GCY0k10jsY

If I use all 4 mods together, I start having issues dragging tokens to move and the darkness spell breaks completely.

If I disable Walled Templates everything works just fine.

I suspect some of these issues are not solely on Walled Templates but also the templates created by Gambit's to track attacks of opportunities. I figured I would post here since this issue sounds very similar to my own.

Versions:
Dnd 5e 3.1.2
Walled Templates 7.10
Chris's 0.10.73
Gambit's 0.1.41
Drag Ruler 1.13.8

**Comment by chrisk123999** [comment]
I think you'll want to use the legacy setting in GPS. As this is likely caused from something Gambit's Premades does, you may want to make an issue with that module.

**Comment by NdranC** [comment]
> I think you'll want to use the legacy setting in GPS. As this is likely caused from something Gambit's Premades does, you may want to make an issue with that module.

So I have uninstalled Drag Ruler and replaced it with the Elevator Ruler and the dragging issues seem to be solved. But that is not why I made a post over here. Darkness still doesn't work when I have Walled Templates enabled. When I try to place it just like in the video nothing happens. I can see a template on the map but the macro doesn't run and I get a ton of errors in the console

![image](https://github.com/chrisk123999/chris-premades/assets/27837113/b8e06ecf-1d9f-4640-ac6e-d0e92cfcc816)
Like so

**Comment by chrisk123999** [comment]
That error doesn't appear to involve CPR.

**Comment by NdranC** [comment]
You are right, I was getting several errors when I enabled Walled Templates and not all were related to each other or from the same mods/settings. Now that I cleared the dragging issue I can see the error seem to be walled templates/midi-qol. I tested with a basic spell that was not modified by CPR.

Is walled templates a recommended a commonly used mod in the midi-qol ecosystem? I'm trying to hide the templates around Gambit's attacks of opportunity and this was recommended as a way to deal with it but I've had a lot of issues with it.
